tests/e2e/cucumber/steps/app-files/resource.ts

```diff
@@ -170,3 +120,60 @@ When(
     }
   }
 )
+
+export const processDownload = async (
+  stepTable: DataTable,
+  pageObject: any,
+  actionType: string
+) => {
+  let downloads
+  let files, parentFolder
+  const downloadInfo = stepTable.hashes().reduce((acc, stepRow) => {
+    const { resource, from } = stepRow
+
+    if (!acc[from]) {
+      acc[from] = []
+    }
+
+    acc[from].push(resource)
+
+    return acc
+  }, {})
+
+  for (const folder of Object.keys(downloadInfo)) {
+    files = downloadInfo[folder]
+    parentFolder = folder
+    downloads = await pageObject.download({
+      folder,
+      names: files,
```

Comment:
> yes if there's no `from` column in data table

I would suggest you to do that here.
```suggestion
    parentFolder = folder !== 'undefined' ? folder : null
    downloads = await pageObject.download({
      folder: parentFolder,
      names: files,
```